### PR TITLE
Add --restore-date param

### PR DESCRIPTION
**Pull request by hyc**
Estimate restore height from given date

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -2947,6 +2948,27 @@ static bool might_be_partial_seed(const epee::wipeable_string &words)
   return seed.size() < 24;
 }
 //----------------------------------------------------------------------------------------------------
+static bool datestr_to_int(std::string heightstr, uint16_t &year, uint8_t &month, uint8_t &day)
```
Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp]: const std::string&

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -2947,6 +2948,27 @@ static bool might_be_partial_seed(const epee::wipeable_string &words)
...
+  {
+    fail_msg_writer() << tr("date format must be YYYY-MM-DD");
+    return false;
+  }
+  try
+  {
+    year  = boost::lexical_cast<uint16_t>(heightstr.substr(0,4));
+    // lexical_cast<uint8_t> won't work because uint8_t is treated as character type
+    month = boost::lexical_cast<uint16_t>(heightstr.substr(5,2));
+    day   = boost::lexical_cast<uint16_t>(heightstr.substr(8,2));
```
Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp]: Neither this nor get_blockchain_height_by_...does much range checking, but I guess it's existing code.

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
Comment (hyc) [reply src/simplewallet/simplewallet.cpp]: ok

File: src/simplewallet/simplewallet.cpp (hunk 2 above)
Comment (hyc) [reply src/simplewallet/simplewallet.cpp]: get_blockchain_height_by_... checks that month and day are sane. Year is pretty open-ended.

**Pull request merged**